src/cryptoadvance/specter/wallet.py

```diff
@@ -307,22 +308,35 @@ def save_pending_psbt(self, psbt):
         self.cli.lockunspent(False, psbt["tx"]["vin"])
         self.save_to_file()
 
-    def txlist(self, idx, wallet_tx_batch=100):
+    def txlist(self, idx, wallet_tx_batch=100, validate_merkle_proofs=False):
         try:
             cli_txs = self.cli.listtransactions("*", wallet_tx_batch + 2, wallet_tx_batch * idx, True) # get batch + 2 to make sure you have information about send
             cli_txs.reverse()
             transactions = cli_txs[:wallet_tx_batch]
         except:
             return []
-        txids = []
         result = []
+        blocks = {}
         for tx in transactions:
             if 'confirmations' not in tx:
                 tx['confirmations'] = 0
             if len([_tx for _tx in cli_txs if (_tx['txid'] == tx['txid'] and _tx['address'] == tx['address'])]) > 1:
                 continue # means the tx is duplicated (change), continue
 
-            txids.append(tx["txid"])
+            tx['validated_blockhash'] = ""  # default is assume unvalidated
+            if validate_merkle_proofs is True:
+                proof_hex = self.cli.gettxoutproof([tx['txid']])
```

Comment:
This can cause an exception for unconfirmed transactions and for confirmed transactions in pruned nodes when the block is pruned.

Reply:
Good point re unconfirmed TXs, but are you sure re pruned nodes? I just updated my PR to add the blockhash, which *should* always work:

https://bitcoincore.org/en/doc/0.20.0/rpc/blockchain/gettxoutproof/
```
NOTE: By default this function only works sometimes. This is when there is an
unspent output in the utxo for this transaction. To make it always work,
you need to maintain a transaction index, using the -txindex command line option or
specify the block in which the transaction is included manually (by blockhash).
```

I'm trying to test a pruned node, but I accidentally triggered a rescan so mine is out for a while :(

Reply:
Actually it doesn't even if there are unspent outputs. That's the reason we had to fetch proofs from blockexplorer in #344  
Just checked on my pruned node, getting an rpc error, even though this transaction is imported to the wallet with the proof.
```
error code: -32603
error message:
Can't read block from disk
```

Reply:
Ah, didn't realize the intricacies of pruned nodes. I assumed they kept all blocks that they had transactions in (or at least the merkle proofs). I wonder what the thought process was there?

I'll update the PR with disabling it altogether for pruned nodes.